### PR TITLE
DBZ-2541 Make log4j root loggers configurable in connect-base image

### PR DESCRIPTION
**Pull request by jfinzel**
https://issues.redhat.com/browse/DBZ-2541

### CODE REVIEW

**Comment by jpechane** [comment]
Hi, thanks for the PR. Fo start it looks good! Please just update the commit message to refer to JIRA id.
IMHO it is fine to use `CONNECT_` prefix. But as these env vars are passed to the Kafka Connect I belive it makes sense to unset all `CONNECT_LOG4J_` env vars after they are used.

File: connect-base/1.3/docker-entrypoint.sh [hunk 1]
```diff
@@ -164,6 +164,28 @@ case $1 in
...
+          prop_name=`echo "$VAR" | sed -r "s/^CONNECT_(.*)=.*/\1/g" | tr '[:upper:]' '[:lower:]' | tr _ .`
+          if egrep -q "(^|^#)$prop_name=" $KAFKA_HOME/config/log4j.properties; then
+              #note that no config names or values may contain an '@' char
+              sed -r -i "s@(^|^#)($prop_name)=(.*)@\2=${!env_var}@g" $KAFKA_HOME/config/log4j.properties
+          else
+              #echo "Adding property $prop_name=${!env_var}"
+              echo "$prop_name=${!env_var}" >> $KAFKA_HOME/config/log4j.properties
+          fi
+          if [[ "$SENSITIVE_PROPERTIES" = *"$env_var"* ]]; then
+              echo "--- Setting property from $env_var: $prop_name=[hidden]"
```
Comment (jpechane) [inline connect-base/1.3/docker-entrypoint.sh:184]: Here it probably make sense to emphasize that this is *logging* property.

File: connect-base/1.3/docker-entrypoint.sh [hunk 2]
```diff
@@ -164,6 +164,28 @@ case $1 in
         #
         # Configure the log files ...
         #
+        if [[ -n "$CONNECT_LOG4J_LOGGERS" ]]; then
+            sed -i -r -e "s|^(log4j.rootLogger)=.*|\1=${CONNECT_LOG4J_LOGGERS}|g" $KAFKA_HOME/config/log4j.properties
+            if [[ "$CONNECT_LOG4J_LOGGERS" != *"appender"* ]]; then
```
Comment (jpechane) [inline connect-base/1.3/docker-entrypoint.sh:169]: Do we really wnt to remove the appender? What if we configure a non-root logger that would use it.

Comment (jfinzel) [reply connect-base/1.3/docker-entrypoint.sh:169]: @jpechane yes, so far as I can tell, the appender is a problem for us.  Here is my issue.  Our standard logging uses stdout and send logs to splunk.  Those log files appear in `/var/log/docker-compose.log` and are managed automatically with docker-compose setting I mentioned with `logging`:
```
    logging:
        driver: "json-file"
        options:
            max-file: "5"
            max-size: "10m"
```
This manages log rotation and compression.  The problem is that `appender` also creates duplicate log files in the docker directories `/var/lib/docker` with names of `connect-service.log.*`.  We have no way to manage these (compress/log rotate), and they are creating out of disk problems for us as a result.  We don't understand the need for those files at all.

If you have any other suggestion I don't know about, I'm open to it.  But we don't see the need at all for `appender` in our use, unless I'm missing something.

Comment (jpechane) [reply connect-base/1.3/docker-entrypoint.sh:169]: But is not it enough to remove appender from root logger but still keep appender config? IMHO unless it is used the config is harmless

Comment (jfinzel) [reply connect-base/1.3/docker-entrypoint.sh:169]: That's right.  It was just for cleanliness, so it doesn't matter either way to me.  Would you rather I remove that part?

Comment (jpechane) [reply connect-base/1.3/docker-entrypoint.sh:169]: Yeah, 'd prefer to keep the appender definition even it is not used as it might be handy for somebody.

**Comment by jfinzel** [comment]
> Hi, thanks for the PR. Fo start it looks good! Please just update the commit message to refer to JIRA id.
> IMHO it is fine to use `CONNECT_` prefix. But as these env vars are passed to the Kafka Connect I belive it makes sense to unset all `CONNECT_LOG4J_` env vars after they are used.

@jpechane I'm not sure I understand regarding unsetting those env vars.  This script does not appear to unset the other `CONNECT_` vars after use (https://github.com/debezium/docker-images/blob/a71620eb9cefcdfa5c6f3101464c9e6bd394897f/connect-base/1.3/docker-entrypoint.sh#L195-L216)?

**Comment by jpechane** [comment]
@jfinzel The start script takes all `CONNECT_` named properties and converts them into Kafka Connect configuration options (adding to the config file). If `CONNECT_LOGJ_` env vars will stary set then the Kafka COnnect configuration will be full of config options named `log4j.*` which I consider to be confusing

**Comment by jfinzel** [comment]
> @jfinzel The start script takes all `CONNECT_` named properties and converts them into Kafka Connect configuration options (adding to the config file). If `CONNECT_LOGJ_` env vars will stary set then the Kafka COnnect configuration will be full of config options named `log4j.*` which I consider to be confusing

@jpechane I follow.  I have pushed the fix for this.  Let me know if there is anything else.

**Comment by jpechane** [comment]
@jfinzel I've added few changes to make it work. Could you please add this change to the docs? - see https://debezium.io/documentation/reference/1.3/operations/logging.html#configuring-log-level-docker

**Comment by jfinzel** [comment]
> @jfinzel I've added few changes to make it work. Could you please add this change to the docs? - see https://debezium.io/documentation/reference/1.3/operations/logging.html#configuring-log-level-docker

https://github.com/debezium/debezium/pull/1850

**Pull request merged**